web/src/components/Visualization/components/DAG/Actions.tsx

```diff
@@ -43,13 +43,17 @@ const Actions = ({
   const handleOnNavigateToSpan = useCallback(
     (spanId: string) => {
       onNavigateToSpan(spanId);
+    },
+    [onNavigateToSpan]
+  );
 
+  useEffect(() => {
+    if (selectedSpan) {
       const {nodeInternals} = getState();
-      const {x, y} = getNodePosition(spanId, nodeInternals);
+      const {x, y} = getNodePosition(selectedSpan, nodeInternals);
       setCenter(x, y, {zoom: 1.0, duration: 1000});
```

Comment:
Now, this updates every time there is a newly selected span.

Reply:
Looks good, not a fan of having this type of things in effects. I prefer to control them using user events.